gradio/components.py

```diff
@@ -4130,34 +4243,65 @@ def update(
         }
         return updated_config
 
-    def _process_chat_messages(
-        self, chat_message: str | Tuple | List | Dict | None
+    def _preprocess_chat_messages(
+        self, chat_message: str | Dict | None
+    ) -> str | Tuple | None:
+        if chat_message is None:
+            return None
+        elif isinstance(chat_message, dict):
+            if chat_message["alt_text"] is not None:
+                return (chat_message["name"], chat_message["alt_text"])
+            else:
+                return (chat_message["name"],)
+        else:  # string
+            return chat_message
+
+    def preprocess(
+        self,
+        y: List[Tuple[str | Dict | None, str | Dict | None]],
+    ) -> List[Tuple[str | Tuple | None, str | Tuple | None]]:
+        if y is None:
+            return y
+        processed_messages = []
+        for message_pair in y:
```

Comment:
The addition of `preprocess` is breaking the `chatbot_multimodal` demo as it changes the types from list to tuples. I think we should be returning list of lists to avoid breaking old demos.

For context, this is the stacktrace

```bash
    prediction = await anyio.to_thread.run_sync(
  File "/Users/freddy/miniconda3/envs/gradio/lib/python3.8/site-packages/anyio/to_thread.py", line 31, in run_sync
    return await get_asynclib().run_sync_in_worker_thread(
  File "/Users/freddy/miniconda3/envs/gradio/lib/python3.8/site-packages/anyio/_backends/_asyncio.py", line 937, in run_sync_in_worker_thread
    return await future
  File "/Users/freddy/miniconda3/envs/gradio/lib/python3.8/site-packages/anyio/_backends/_asyncio.py", line 867, in run
    result = context.run(func, *args)
  File "demo/chatbot_multimodal/run.py", line 13, in bot
    response = "**That's cool!**"
TypeError: 'tuple' object does not support item assignment
```


![image](https://user-images.githubusercontent.com/41651716/226448815-70f6faf5-2f42-4fc9-8ea9-cf76c5bb4b66.png)

Reply:
Changed to lists so now the demo is working, thanks for catching that @freddyaboulton!